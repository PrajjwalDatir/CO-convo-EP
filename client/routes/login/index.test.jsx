import React from 'react';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router-dom';
import Login from './index';

describe('<Login />', () => {
  it('should match snapshot', () => {
    const wrapper = mount(
      <StaticRouter>
        <Login />
      </StaticRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a username field', () => {
    const wrapper = mount(
      <StaticRouter>
        <Login />
      </StaticRouter>,
    );
    const username = wrapper.find('input[name="username"]');
    expect(username).toHaveLength(1);
  });

  it('should have a password field', () => {
    const wrapper = mount(
      <StaticRouter>
        <Login />
      </StaticRouter>,
    );
    const password = wrapper.find('input[type="password"][name="password"]');
    expect(password).toHaveLength(1);
  });

  it('should have a login button', () => {
    const wrapper = mount(
      <StaticRouter>
        <Login />
      </StaticRouter>,
    );
    const button = wrapper.find('button[type="submit"]');
    expect(button).toHaveLength(1);
  });

  it('should update the username field', () => {
    const wrapper = mount(
      <StaticRouter>
        <Login />
      </StaticRouter>,
    );

    wrapper
      .find('input[name="username"]')
      .simulate('change', { target: { value: 'some-username' } });

    expect(wrapper.find('input[name="username"]').prop('value')).toBe(
      'some-username',
    );
  });

  it('should update the password field', () => {
    const wrapper = mount(
      <StaticRouter>
        <Login />
      </StaticRouter>,
    );

    wrapper
      .find('input[type="password"][name="password"]')
      .simulate('change', { target: { value: 'some-password' } });

    expect(
      wrapper.find('input[type="password"][name="password"]').prop('value'),
    ).toBe('some-password');
  });
});
