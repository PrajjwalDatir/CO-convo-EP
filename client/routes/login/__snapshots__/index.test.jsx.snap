// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Login>
      <DottedBackground
        className=""
        style={Object {}}
      >
        <div
          className="h-screen flex flex-col justify-center items-center "
          style={
            Object {
              "backgroundImage": "url(test-file-stub)",
            }
          }
        >
          <LoginCardForm
            buttonText="Login"
            header="Login"
            image="test-file-stub"
            onSubmit={[Function]}
          >
            <Card
              className="overflow-x-hidden"
            >
              <div
                className="rounded bg-white p-10 shadow-lg overflow-x-hidden"
              >
                <div
                  className="flex w-full"
                >
                  <styled.div>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "jBDajD",
                            "rules": Array [
                              "
  width: 300px;
",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="sc-bdVaJa jBDajD"
                      >
                        <h1
                          className="text-4xl font-light mb-4"
                        >
                          Login
                        </h1>
                        <form
                          onSubmit={[Function]}
                        >
                          <LoginCardFormField
                            errors={Object {}}
                            label="Username"
                            name="username"
                            onChange={[Function]}
                            type="username"
                            validation={[Function]}
                            value=""
                          >
                            <div
                              key="username"
                            >
                              <label
                                htmlFor="username-field"
                              >
                                <div
                                  className="flex"
                                >
                                  <div
                                    className="mb-1 uppercase font-bold tracking-wide text-xs text-gray-600"
                                  >
                                    Username
                                  </div>
                                  <div
                                    className="text-xs text-red-700 ml-auto"
                                  />
                                </div>
                                <ForwardRef
                                  className="w-full mb-4"
                                  id="username-field"
                                  name="username"
                                  onChange={[Function]}
                                  type="username"
                                  value=""
                                >
                                  <input
                                    className="rounded-lg bg-gray-200 px-3 py-2 border border-solid border-transparent hover:border-gray-400 focus:bg-white focus:border-gray-400 w-full mb-4"
                                    id="username-field"
                                    name="username"
                                    onChange={[Function]}
                                    type="username"
                                    value=""
                                  />
                                </ForwardRef>
                              </label>
                            </div>
                          </LoginCardFormField>
                          <LoginCardFormField
                            errors={Object {}}
                            label="Password"
                            name="password"
                            onChange={[Function]}
                            type="password"
                            validation={[Function]}
                            value=""
                          >
                            <div
                              key="password"
                            >
                              <label
                                htmlFor="password-field"
                              >
                                <div
                                  className="flex"
                                >
                                  <div
                                    className="mb-1 uppercase font-bold tracking-wide text-xs text-gray-600"
                                  >
                                    Password
                                  </div>
                                  <div
                                    className="text-xs text-red-700 ml-auto"
                                  />
                                </div>
                                <ForwardRef
                                  className="w-full mb-4"
                                  id="password-field"
                                  name="password"
                                  onChange={[Function]}
                                  type="password"
                                  value=""
                                >
                                  <input
                                    className="rounded-lg bg-gray-200 px-3 py-2 border border-solid border-transparent hover:border-gray-400 focus:bg-white focus:border-gray-400 w-full mb-4"
                                    id="password-field"
                                    name="password"
                                    onChange={[Function]}
                                    type="password"
                                    value=""
                                  />
                                </ForwardRef>
                              </label>
                            </div>
                          </LoginCardFormField>
                          <Button
                            className="w-full bg-blue-600 hover:bg-blue-700"
                            type="submit"
                          >
                            <button
                              className="px-4 py-2 bg-gray-700 hover:bg-gray-800 text-gray-100 rounded shadow hover:shadow-md focus:shadow-outline cursor-pointer w-full bg-blue-600 hover:bg-blue-700"
                              type="submit"
                            >
                              Login
                            </button>
                          </Button>
                        </form>
                      </div>
                    </StyledComponent>
                  </styled.div>
                  <div
                    className="ml-12 -mr-20"
                  >
                    <img
                      alt=""
                      className="w-64"
                      src="test-file-stub"
                    />
                  </div>
                </div>
              </div>
            </Card>
          </LoginCardForm>
        </div>
      </DottedBackground>
    </Login>
  </Router>
</StaticRouter>
`;
